src/silo/preprocessing/preprocessing_database.cpp

```diff
@@ -30,97 +31,26 @@ using duckdb::StringVector;
-      BinaryExecutor::Execute<string_t, string_t, string_t>(
-         args.data[0],
-         args.data[1],
-         result,
-         args.size(),
-         [&](const string_t uncompressed, const string_t segment_name) {
-            const std::string_view compressed =
-               nuc_compressors.at(segment_name.GetString())
-                  .local()
-                  .compress(uncompressed.GetData(), uncompressed.GetSize());
-            return StringVector::AddStringOrBlob(
-               result, compressed.data(), static_cast<uint32_t>(compressed.size())
-            );
-         }
-      );
-   };
-
-   static void compressAA(DataChunk& args, ExpressionState& /*state*/, Vector& result) {
-      BinaryExecutor::Execute<string_t, string_t, string_t>(
-         args.data[0],
-         args.data[1],
-         result,
-         args.size(),
-         [&](const string_t uncompressed, const string_t gene_name) {
-            const std::string_view compressed =
-               aa_compressors.at(gene_name.GetString())
-                  .local()
-                  .compress(uncompressed.GetData(), uncompressed.GetSize());
-            return StringVector::AddStringOrBlob(
-               result, compressed.data(), static_cast<uint32_t>(compressed.size())
-            );
-         }
-      );
-   }
-};
-
-std::unordered_map<std::string_view, tbb::enumerable_thread_specific<silo::ZstdCompressor>>
-   Compressors::nuc_compressors{};
-std::unordered_map<std::string_view, tbb::enumerable_thread_specific<silo::ZstdCompressor>>
-   Compressors::aa_compressors{};
-
-}  // namespace
-
 namespace silo::preprocessing {
 
 PreprocessingDatabase::PreprocessingDatabase(
-   const std::optional<std::filesystem::path>& backing_file
+   const std::optional<std::filesystem::path>& backing_file,
+   const ReferenceGenomes& reference_genomes
 )
-    : duck_db(backing_file.value_or(":memory:")),
+    : compress_nucleotide_function(
+         std::make_unique<CompressSequence>("nuc", reference_genomes.raw_nucleotide_sequences)
+      ),
+      compress_amino_acid_function(
+         std::make_unique<CompressSequence>("aa", reference_genomes.raw_aa_sequences)
+      ),
+      duck_db(backing_file.value_or(":memory:")),
       connection(duck_db) {
    query("PRAGMA default_null_order='NULLS FIRST';");
    query("SET preserve_insertion_order=FALSE;");
    query("SET memory_limit='50 GB';");
 
-   connection.CreateVectorizedFunction(
-      std::string(COMPRESS_NUC),
-      {LogicalType::VARCHAR, LogicalType::VARCHAR},
-      LogicalType::BLOB,
-      Compressors::compressNuc
-   );
-
-   connection.CreateVectorizedFunction(
-      std::string(COMPRESS_AA),
-      {LogicalType::VARCHAR, LogicalType::VARCHAR},
-      LogicalType::BLOB,
-      Compressors::compressAA
-   );
+   compress_nucleotide_function->applyTo(connection);
+   compress_amino_acid_function->applyTo(connection);
```

Comment:
`applyTo` seems a little unintuitive here, I think?

Reply:
I am open for suggestions. How about 'addToConnection'?